### ISSUE TITLE
Hue Light Switches

### ISSUE THREAD

**Issue by lespatots** [Foddy/node-red-contrib-huemagic#16]
Is it possible to add a node to provide the last setting/press of tap and dimmer switches. As far as I can tell this ought to be possible and is the one piece of missing Hue functionality that would make the switches finally really useful? I know it is possible to follow the state of a dependant light but that is an unnecessary complication.

Change detection would be ideal but (for me) polling of switches is the next best alternative and it would be possible to locate all the switches on their own bridge to limit the affect on the overall system.

**Comment by Foddy** [comment]
Hi lespatots,

sure! It’s a great idea. Honestly, I already planned the support of Hue Switches by HueMagic in near future. Probably with the next update if everything works well. I’ll let you know when it’s ready ;)

**Comment by Foddy** [comment]
Hue Tap and Switch devices are now supported with v1.2.1… ;)
Let me know if they are not working properly since I don't own any of these devices

Best,
Foddy

**Comment by lespatots** [comment]
Foddy,

I’ve been looking further at my Hue Switch problem and I’m not convinced all is rosy in the land of Huejay. I have two Pi’s running Node-Red, One is using the old Hue plus nodes whilst the other has your new nodes. The first offered an upgrade to your 1.2.1 version which I upgraded (but don’t use). The other didn’t and I had to use NPM to do an install which as I said generated some WARN messages. I just compared the Huejay part of the directory and they are not the same in the Node Modules directory. Apart from completely re-installing Node-red is there a way to ensure Huejay is correct?

Any help much appreciated. Just to note that having the two instances of Node-Red lets me run both types of Hue Nodes simultaneously (or may be doesn’t and I’m now seeing the result?)

Any help much appreciated.

Richard
